matchbox_server/src/signaling.rs

```diff
@@ -453,8 +463,159 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn match_different_id_same_next() {
```

Comment:
here we check
- player a -> scope_1?next=2
- player b -> scope_2?next=2
- player c -> scope_1?next=2
- player d -> scope_2?next=2